lib/internal/modules/esm/loader.js

```diff
@@ -388,7 +396,46 @@ function createModuleLoader(useCustomLoadersIfPresent = true) {
   return new DefaultModuleLoader();
 }
 
+/**
+ * Get the HooksProxy instance. If it is not defined, then create a new one.
+ * @returns {HooksProxy}
+ */
+function getHooksProxy() {
+  if (!hooksProxy) {
+    const { HooksProxy } = require('internal/modules/esm/hooks');
+    hooksProxy = new HooksProxy();
+  }
+
+  return hooksProxy;
+}
+
+/**
+ * Register a single loader programmatically.
+ * @param {string} specifier
+ * @param {string} [parentURL]
+ * @returns {void}
+ * @example
+ * ```js
+ * register('./myLoader.js');
+ * register('ts-node/esm', import.meta.url);
+ * register('./myLoader.js', import.meta.url);
+ * register(new URL('./myLoader.js', import.meta.url));
+ * ```
+ */
+function register(specifier, parentURL = 'file:///') {
```

Comment:
We should probably default to a URL that do no support relative URLs at all, so we're consistent cross platform. I've given some thoughts about setting `file:///c/` on Windows, but that seems like a something that would no be obvious at all, so using a different "neutral" protocol seems safer.
```suggestion
function register(specifier, parentURL = 'data:') {
```